Source/Diagnostics/ComputeDiagFunctors/JFunctor.cpp

```diff
@@ -0,0 +1,70 @@
+/* This file is part of WarpX.
+ *
+ * Authors: Roelof Groenewald
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "JFunctor.H"
+
+#include "Particles/MultiParticleContainer.H"
+#include "WarpX.H"
+
+#include <AMReX.H>
+#include <AMReX_IntVect.H>
+#include <AMReX_MultiFab.H>
+
+JFunctor::JFunctor (const int dir, int lev,
+                   amrex::IntVect crse_ratio,
+                   bool convertRZmodes2cartesian,
+                   bool deposit_current, int ncomp)
+    : ComputeDiagFunctor(ncomp, crse_ratio), m_dir(dir), m_lev(lev),
+      m_convertRZmodes2cartesian(convertRZmodes2cartesian),
+      m_deposit_current(deposit_current)
+{ }
+
+void
+JFunctor::operator() (amrex::MultiFab& mf_dst, int dcomp, const int /*i_buffer*/) const
+{
+    auto& warpx = WarpX::GetInstance();
+    /** pointer to source multifab (can be multi-component) */
+    amrex::MultiFab* m_mf_src = warpx.get_pointer_current_fp(m_lev, m_dir);
+
+    // Deposit current if no solver or the electrostatic solver is being used
+    if (m_deposit_current)
+    {
+        // allocate temporary multifab to deposit current density into
+        amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > > current_fp_temp;
+        current_fp_temp.resize(1);
+
+        auto& current_fp_x = warpx.getcurrent_fp(m_lev,0);
+        current_fp_temp[0][0] = std::make_unique<amrex::MultiFab>(
```

Comment:
This should work instead of creating temporary MultiFabs and then copying.
```
current_fp_temp[0][0] = std::make_unique<amrex::MultiFab>(current_fp_x, amrex::make_alias, 0, 1);
```
I tested this and it at least compiles.

Reply:
Thanks! I didn't know about the `make_alias` functionality. That is great to learn about and I'm sure it'll come in handy again in the future.